docs/architecture.md

```diff
@@ -82,17 +94,32 @@ def my_command(ctx, pulp_ctx):
     ctx.obj = MyEntityContext(pulp_ctx)
 ```
 
-The named tuple `PluginRequirement` is used to describe dependence on server components.
-It needs the `name` of the plugin and accepts optionally an inclusive `min` version,
-an exclusive `max` version, and an `inverted` flag for exclusion of a version range.
+To declare version restrictions on *options* (*body parameters* in the API), the [`preprocess_entity`][pulp_glue.common.context.PulpEntityContext.preprocess_entity]
+method can check if a given option is being used and conditionally apply the requirements to the context.
```

Comment:
This function is intended more generally. We also fix some inputs there or perform transformations on certain fields.

Reply:
The sentences says `preprocess_entity` can be used to achieve a task, but it doesn't imply that it's the only thing it can be used for (and the linked reference description also enforces its  general purpose) nature. Given the context ("Version dependent path" section), I think it's reasonable to just mention this one. Wdyt?

Reply:
Sounds good too.
And we can always reiterate...